File: scala-spring/accounts-query-side-backend/src/main/scala/net/chrisrichardson/eventstore/examples/bank/queryside/QuerysideDependencyChecker.scala
package net.chrisrichardson.eventstore.examples.bank.queryside

import javax.annotation.PostConstruct

import net.chrisrichardson.utils.logging.Logging
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.mongodb.core.MongoTemplate

import scala.concurrent.{TimeoutException, Await, Future}
import scala.concurrent.duration._
import scala.concurrent.ExecutionContext.Implicits.global

class QuerysideDependencyChecker (mongoTemplate : MongoTemplate) extends Logging {

  @PostConstruct
  def checkDependencies(): Unit = {
    try {
      Await.result(Future { mongoTemplate.getDb.getCollectionNames}, 5 seconds)
    } catch {
      case e : Throwable =>
        logger.error("Error connecting to Mongo - have you set SPRING_DATA_MONGODB_URI or --spring.data.mongodb_uri?", e)
        throw e
    }
  }

}

File: java-spring/accounts-view-service/src/main/java/net/chrisrichardson/eventstore/javaexamples/banking/accountsviewservice/backend/QuerySideDependencyChecker.java
package net.chrisrichardson.eventstore.javaexamples.banking.accountsviewservice.backend;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.data.mongodb.core.MongoTemplate;

import javax.annotation.PostConstruct;

public class QuerySideDependencyChecker {
  private Logger logger = LoggerFactory.getLogger(getClass());
  private MongoTemplate mongoTemplate;

  public QuerySideDependencyChecker(MongoTemplate mongoTemplate) {
    this.mongoTemplate = mongoTemplate;
  }

  @PostConstruct
  public void checkDependencies() {
    try {
      logger.info("Checking mongodb connectivity {}", System.getenv("SPRING_DATA_MONGODB_URI"));

      mongoTemplate.getDb().getCollectionNames();

    } catch (Throwable e) {
      throw new RuntimeException("Error connecting to Mongo - have you set SPRING_DATA_MONGODB_URI or --spring.data.mongodb_uri?", e);
    }
  }
}
